### ISSUE TITLE
Allow explicit module paths' data to override data from Python packages

### ISSUE THREAD

**Issue by ricklupton** [probs-lab/probs-runner#13]
Currently scripts are overridden by explicit paths, but data is not: https://github.com/probs-lab/probs-runner/blob/master/src/probs_runner/runners.py#L94-L128

The order should be swapped.

**Comment by stephenjboyle** [comment]
@ricklupton I think this commit will implement this:
https://github.com/probs-lab/probs-runner/commit/bfbfb13120e654c3167ab90fd20fab31fe0c96a2

**Comment by ricklupton** [comment]
Thanks @stephenjboyle. I think it would be better to check for each file, not ignore installed modules completely as soon as one file from the explicit path is found.

Something like replacing this

```python
    # Use the version of the data bundled with the Python package, if available
    try:
        data_source_dir = importlib_resources_files("probs_system.data")
    except ModuleNotFoundError:
        pass
    else:
        # Need to add data files individually by discovering which are available
        # (the MultiplexedPath from importlib.resources cannot be directly copied)
        if not isinstance(data_source_dir, Path):
            for path in data_source_dir._paths:
                for p in path.iterdir():
                    rel = Path("data") / p.relative_to(path)

                    # FIXME: remove when RDFox supports gzip on Windows.
                    # Work around [lack of] compression by RDFox: if needed,
                    # make a copy of the data file that's [de]compressed
                    p = prepare_file_for_rdfox(p, rel)

                    input_files[rel] = p
        else:
            input_files["data"] = data_source_dir
 
```

with this


```python
    # Use the version of the data bundled with the Python package, if available,
    # but don't overwrite files from explicit path above.
    try:
        data_source_dir = importlib_resources_files("probs_system.data")
    except ModuleNotFoundError:
        pass
    else:
        # Need to add data files individually by discovering which are available
        # (the MultiplexedPath from importlib.resources cannot be directly copied)
        if not isinstance(data_source_dir, Path):
            paths = data_source_dir._paths
        else:
            paths = [data_source_dir]
        for path in paths:
            for p in path.iterdir():
                rel = Path("data") / p.relative_to(path)

                # FIXME: remove when RDFox supports gzip on Windows.
                # Work around [lack of] compression by RDFox: if needed,
                # make a copy of the data file that's [de]compressed
                p = prepare_file_for_rdfox(p, rel)

                if rel not in input_files:
                    input_files[rel] = p
```

might do?

**Comment by stephenjboyle** [comment]
@ricklupton 
Sorry, I misunderstood what was required. I thought any explicit data folder (containing files) was to override data in Python packages. I've changed the code in commit https://github.com/probs-lab/probs-runner/commit/6cf626c9085f5e8073c017b6e50f46ae36211315 and tested it. Seems to work ok. I had to change it slightly or it was still adding duplicate files from packages because the keys used in ```input_files``` from the two sources were of different types, one was Posix Path and the other a string. Changed it so it always uses a  string for the key.
